dataset-processor-python/data_loader.py

```diff
@@ -31,25 +43,186 @@ class DatasetWriter(DataLoader):
 
     def __init__(self, path, dataset_name):
         super().__init__(path, dataset_name)
+        self.cell_feature_analysis_file = []
+        self.feature_defs_file = []
+        self.dataset_file = {}
+        self.feature_def_keys = []
+        self.discrete_features_dict = {}
+        self.features_data_order = []
```

Comment:
Nit: Might be nice to have additional comments on these explaining the expected key/value types? Especially for the dictionaries, like `dataset_file` and `discrete_features_dict`.